lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+
+/**
+ * Html file representation (needs for errors output).
+ *
+ * @name HtmlFile
+ */
+var HtmlFile = function(filename, source) {
+    this._filename = filename;
+    this._lines = source.split(/\r\n|\r|\n/);
+};
+
+HtmlFile.prototype = {
+    /**
+     * Returns source filename for this object representation.
+     *
+     * @returns {String}
+     */
+    getFilename: function() {
+        return this._filename;
+    },
+    /**
+     * Returns array of source lines for the file.
+     *
+     * @returns {String[]}
+     */
+    getLines: function() {
+        return this._lines;
+    }
+};
+
+/**
+ * Parse html and retrieve script sources.
+ * @param {String} html
+ * @returns {Object[]}
+ */
+function getScripts(html) {
+    function onopen(name, attrs) {
```

Comment:
Ah, I see that this depends on closed over vars later on. Tiny touch confusing, but no harm done. A class might be clearer, but not a blocker.